### DISCUSSION TITLE
C7n-org dry run using wrong role for periodic/CloudTrail policies?

### DISCUSSION THREAD

**Discussion by danielhros** [cloud-custodian/cloud-custodian#9698] [AWS]
In my setup, I am deploying periodic and cloudtrail policies exclusively. Before deployment, I run c7n-org with the `--dryrun` flag. I noticed that c7n-org is executing policies using the role specified in the accounts.yml configuration file instead of the role specified in the policy.

What's the point of using the IAM role from the accounts.yml file when that role is only used for the deployment of lambda functions, and then the lambda function itself is using a different IAM role? I would expect to use the IAM role specified in the Cloud Custodian policy instead of the one from the accounts.yml file so that if the IAM role doesn't have enough permissions, we would catch that before deployment. Or am I missing something?

**Comment by ajkerrigan** [comment]
Dry run mode is mostly about catching policy logic issues - it's similar to running a policy with the `mode` and `actions` blocks removed. It's admittedly an awkward fit for event-based policies.

**Reply by danielhros** [reply]
Understood. Given that we manage over 400+ accounts, running a dry run before deployment might be time-consuming. Do you think it is still advisable to perform a dry run in this context?
